### ISSUE TITLE
Is there a way to stop rendering when choosing worldmap as the view?

### ISSUE THREAD

**Issue by seanDot7** [apache/incubator-superset#926]
I want to choose world map to visualize my data. But it starts to render with default column, it takes long time. I want to choose the country code field first and then render the data. I don't know how to do it?

**Comment by xrmx** [comment]
@seanDot7 If the code doesn't do what you need, you have to change the code. There's not shortcuts. If you need code pointers please use the mailing list though instead of opening issues there. Thanks.

**Comment by seanDot7** [comment]
Ok thanks.

**Comment by DylanAlloy** [comment]
@xrmx It's a glitch. It'll recognize the wrong data field (even though I clearly have one labeled "Country") and then the user cannot do anything.

Like the specific indication for temporal data dimensions, there should be one specifically for global data or "spatial". People can add this themselves, sure, but it's disingenuous to say that it's not a hindrance/glitch the way it is now.

**Comment by seanDot7** [comment]
I think there should be an approach to stop rendering. For example if I choose the wrong parameters and I want to stop it and re-render the data. This seems to be an common operation in my caravel app.

**Comment by DylanAlloy** [comment]
@seanDot7 Agreed. I'm going to work on this as it really gets in the way of some cool viz. I also will work on adding better color coding to the world map. You'll notice countries with "very little data" are white (which makes them blend in with the background) and countries with no data at all are grey, making them visible when they are not represented. 

I'll add these features and probably fork it within the month.

**Comment by drorata** [comment]
+1 The current behavior is not helpful as it tried to build the world map using the wrong column.

**Comment by jbwhit** [comment]
+1 I would also like to add my voice to this issue. I've created a 2 minute screencast to demonstrate what's happening here: https://youtu.be/J3xZLPb4L7M

I honestly don't know how to get around it (it's a blocker, not merely an inconvenience) -- even after changing the Country code to be the correct one, I can't get it to re-query.

**Comment by xrmx** [comment]
@jbwhit now i am curious: if it's a blocker for you why don't you invest your time or money to fix it?

The issue here (surprise!) is that long running queries suck: would it matter to kill queries if the query takes 5 seconds? Sometimes the fix for this slowness is easy as adding indexes but you have to profile and see where the time is spent.

**Comment by jbwhit** [comment]
@xrmx The comments above (including yours) seem to suggest that the problem is a matter of taking a few minutes _before it works_. The problem is not that the query takes 5 minutes... (it's an incredibly small and simple query that takes a tenth of a second to return the table view). The problem is that it **never** renders a map. 

Never.

And it's not because of the query size or complexity, so profiling the query is time misspent. Is it a simple fix? Is there a way to simply force the correct Country encoding when the world-map is initially selected? Perhaps! But, I haven't found how to do that in the roughly 4 hours I've spent on this one visualization.

As for spending my own time on it trying to fix it: I plan to. However, I spent last week assessing Caravel: whether we wanted to use it internally and whether we wanted to recommend it to our clients, and I give my report tomorrow.
